staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go

```diff
@@ -241,6 +319,21 @@ func (o *AutoscaleOptions) Run() error {
 	return nil
 }
 
+func (o *AutoscaleOptions) canCreateHPAV1() (bool, error) {
+	// Allow fallback to v1 HPA only if:
+	// 1. CPUPercent is set and Memory is not set.
+	// 2. Or, Memory is not set and the metric type is UtilizationMetricType.
+	var isUtilizationMetricType bool
+	_, _, metricsType, err := parseResourceInput(o.CPU, corev1.ResourceCPU)
+	if err != nil {
+		return false, err
+	}
+	if metricsType == autoscalingv2.UtilizationMetricType {
+		isUtilizationMetricType = true
+	}
+	return (o.CPUPercent >= 0 && o.Memory == "") || (o.Memory == "" && isUtilizationMetricType), nil
```

Comment:
```suggestion
	return (o.CPUPercent >= 0 && o.Memory == "") || (o.Memory == "" && metricsType == autoscalingv2.UtilizationMetricType), nil
```
and you can drop that variable, it's simple enough to read :wink: